script/cfg/apply_security.py

```diff
@@ -771,19 +771,37 @@ def __init__(self, mgmtxml, is_from_yacfg):
             self.namespaces = {'mgmt': "http://activemq.org/schema"}
         else:
             self.namespaces = {'mgmt': "http://activemq.apache.org/schema"}
+        self.connector_keymap = OrderedDict([
+            ('host', 'connector-host'),
+            ('port', 'connector-port'),
+            ('rmiregistryport', 'rmi-registry-port'),
+            ('jmxrealm', 'jmx-realm'),
+            ('objectname', 'object-name'),
+            ('authenticatortype', 'authenticator-type'),
+            ('secured', 'secured'),
+            ('keystoreprovider', 'key-store-provider'),
+            ('keystoretype', 'key-store-type'),
+            ('keystorepath', 'key-store-path'),
+            ('keystorepassword', 'key-store-password'),
+            ('truststoreprovider', 'trust-store-provider'),
+            ('truststoretype', 'trust-store-type'),
+            ('truststorepath', 'trust-store-path'),
+            ('truststorepassword', 'trust-store-password'),
+            ('passwordcodec', 'password-codec')
+        ])
 
     def merge_connector_from(self, new_connector):
         new_connector_str = StringIO()
         new_connector_str.write('<connector ')
-        for prop in ['connector-host', 'connector-port', 'rmi-registry-port', 'jmx-realm', 'object-name',
-                     'authenticator-type', 'secured', 'key-store-provider', 'key-store-type', 'key-store-path',
-                     'key-store-password', 'trust-store-provider', 'trust-store-type', 'trust-store-path',
-                     'trust-store-password', 'password-codec']:
-            something_written = False
-            if prop in new_connector and new_connector[prop] is not None:
-                new_connector_str.write(prop)
+        something_written = False
+        for prop_key in self.connector_keymap:
+            if new_connector.get(prop_key) is not None:
```

Comment:
The loop could be optimized by getting the prop_name(`self.connector_keymap.get(prop_key)`) and the prop_value(`new_connector.get(prop_key)`) one time at the beginning, i.e.
```suggestion
        for prop_key, prop_name in self.connector_keymap:
            prop_value = new_connector.get(prop_key)
            if prop_value is not None:
```

Reply:
that makes sense. Thanks!